stl/src/print.cpp

```diff
@@ -59,25 +59,54 @@ namespace {
     public:
         _Allocated_string() = default;
 
-        explicit _Allocated_string(__crt_unique_heap_ptr<wchar_t>&& _Other_str, const size_t _Other_capacity) noexcept
-            : _Str(_STD move(_Other_str)), _Str_capacity(_Other_capacity) {}
+        ~_Allocated_string() {
+            if (_Using_heap()) {
+                _Str.~__crt_unique_heap_ptr<wchar_t>();
+            }
+        }
 
-        [[nodiscard]] wchar_t* _Data() const noexcept {
-            return _Str.get();
+        [[nodiscard]] wchar_t* _Data() noexcept {
+            return _Using_heap() ? _Str.get() : _Buffer;
         }
 
         [[nodiscard]] size_t _Capacity() const noexcept {
             return _Str_capacity;
         }
 
-        void _Reset() noexcept {
+        bool _Grow(const size_t _Capacity) {
+            if (_Capacity <= _Str_capacity) {
+                return true;
+            }
+
+            // This is actually a reset in std::unique_ptr terms, it deallocates the memory
             _Str.release();
             _Str_capacity = 0;
+
+            __crt_unique_heap_ptr<wchar_t> _Wide_str{_malloc_crt_t(wchar_t, _Capacity)};
+            if (!_Wide_str) [[unlikely]] {
+                return false;
+            }
+
+            _Str          = std::move(_Wide_str);
+            _Str_capacity = _Capacity;
+
+            return true;
         }
 
     private:
-        __crt_unique_heap_ptr<wchar_t> _Str;
-        size_t _Str_capacity = 0;
+        // Allows small formatted strings, such as those from _Print_to_unicode_console_it, to not allocate any extra
+        // internal transcoding buffer
+        static constexpr size_t _Buffer_size = 2048;
+
+        bool _Using_heap() const {

```

Comment:
`noexcept` can be added here and to several places, although this is not so that critical.

Reply:
Added here and to grow